app/controllers/concerns/repo_setup.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module RepoSetup
+  extend ActiveSupport::Concern
+
+  def setup_status(assignment_repo)
+    repo     = assignment_repo.github_repository
+    progress = { status: :importing }
+
+    return progress unless repo.import_progress[:status] == "complete"
+
+    progress.update(status: :configuring) if assignment_repo.configuring?
+    progress.update(status: :complete) if assignment_repo.configured? || !repo.branch_present?("github-classroom")
```

Comment:
Can we move the branch name to an application constant?

I'm not a huge fan of the rando strings hanging around.

```ruby
GitHubClassroom.default_branch
```

Or something of the like